UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,337 @@
+                    if (newValue != targetScript.ViewAdapterTypeName)
+                    {
+                        Undo.RecordObject(targetScript, "Set view adapter options");
+                        targetScript.ViewAdapterOptions = null;
+                    }
+
+                    UpdateProperty(
+                        updatedValue => targetScript.ViewAdapterTypeName = updatedValue,
+                        targetScript.ViewAdapterTypeName,
+                        newValue,
+                        "Set view adapter"
+                    );
+                }
+            );
+
+            Type adapterType;
+            viewAdapterOptionsFade.target = ShouldShowAdapterOptions(targetScript.ViewAdapterTypeName, out adapterType);
+
+            EditorStyles.label.fontStyle = viewAdapterOptionsPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            ShowAdapterOptionsMenu(
+                "View adapter options",
+                adapterType,
+                options => targetScript.ViewAdapterOptions = options,
+                targetScript.ViewAdapterOptions,
+                viewAdapterOptionsFade.faded
+            );
+
+            EditorGUILayout.Space();
+
+            EditorStyles.label.fontStyle = viewModelPropertyPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            var adaptedViewPropertyType = AdaptTypeBackward(viewPropertyType, targetScript.ViewAdapterTypeName);
+            ShowViewModelPropertyMenu(
+                new GUIContent("View-model property", "Property on the view-model to bind to."),
+                TypeResolver.FindBindableProperties(targetScript),
+                updatedValue => targetScript.ViewModelPropertyName = updatedValue,
+                targetScript.ViewModelPropertyName,
+                property => property.PropertyType == adaptedViewPropertyType
+            );
+
+            GUI.enabled = guiPreviouslyEnabled;
+
+            EditorStyles.label.fontStyle = defaultLabelStyle;
+
+            EditorGUILayout.Space();
+
+        }
+
+        private void ShowAnimatorParametersMenu(
+            GUIContent label,
+            Action<AnimatorParameterTypeAndName> propertyValueSetter,
+            AnimatorParameterTypeAndName curPropertyValue,
+            out Type selectedPropertyType
+        )
+        {
+            var animator = targetScript.GetComponent<Animator>();
+
+            AnimatorControllerParameter[] properties = new AnimatorControllerParameter[] { };
+
+            // For whatever reason, accessing Animator.parameters while the GameObject the
+            // Animator is on is inactive causes it to return an empty array, aswell as fill
+            // the unity console with warnings. This behaviour is undocumented, and this is just
+            // a work around which seems to work fine
+            if (!animator.gameObject.activeInHierarchy)
+            {
+                UnityEditorInternal.ComponentUtility.CopyComponent(animator.transform);
```

Comment:
I'm very hesitant to use anything that calls to `UnityEditorInternal` because it is an undocumented API. This means that any Unity release, even one marked as a minor or bugfix one, could potentially remove it, change the way it works breaking our code, or add unintended side effects.

Reply:
I will look more into what else I could do. I think just getting the scale, position, and rotation and the setting those again will achieve the same thing :)

Reply:
Cool, thanks 👍